WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerAdapter.java

```diff
@@ -35,6 +37,8 @@
 
 public class SitePickerAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
 
+    private static final int NOT_DEF = -1;
```

Comment:
I can't say I'm fond of how this constant is used. It seems it appears in at least 3 instances that are totally unrelated to each other:
1. As an indication that the UI is not initialised yet [here](https://github.com/malinajirka/WordPress-Android/blob/152af75eced32ad80ca9f755f47de235551e4c15/WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerAdapter.java#L294)
2. As the indicator when a list can't find an item [here](https://github.com/malinajirka/WordPress-Android/blob/152af75eced32ad80ca9f755f47de235551e4c15/WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerAdapter.java#L333)
3. As the indicator an item is `less than` another when sorting the sites [here](https://github.com/malinajirka/WordPress-Android/blob/152af75eced32ad80ca9f755f47de235551e4c15/WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerAdapter.java#L577)

I can be wrong but, I think that the use of `-1` in all these cases is circumstantial and putting it behind a common constant is confusing and can lead to subtle bugs if the value changes.

Here's what I would recommend instead:
1. Use the numeral (`-1`) directly for the "item not found" and "item comparison" cases. `-1` is normally expected in such APIs as per the documentation. Defining a new constant on top of this probably makes it less clear.
2. Don't rely on a special value (`-1) for an integer variable to denote UI state. Make `mSelectedItemPos` zero-based and holding the position of the selected item _in the dataset_. The initial value (zero) is a valid and expected initial selection anyway. Let me know if this is not clear and I can expand. Thanks!

Reply:
Completely agree! I'm not sure what was I thinking.